CardScan/Classes/API.swift

```diff
@@ -135,6 +135,67 @@ public struct Api {
         apiCall(endpoint: endpoint, parameters: apiParameters, completion: completion)
     }
     
+    static public func getModelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        if baseUrl == nil || apiKey == nil {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+        
+        modelDownloadConfig(endpoint: endpoint, parameters: parameters, completion: completion)
+    }
+    
+    private static func modelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        guard let baseUrl = self.baseUrl else {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+        
+        guard let url = urlWithQueryParameters(baseUrl: baseUrl, endpoint: endpoint, parameters: parameters) else {
+            DispatchQueue.main.async { completion(nil, defaultError) }
+            return
+        }
+
+        let session = URLSession(configuration: configuration())
+        session.dataTask(with: url) { data, response, error in
+            guard let rawData = data else {
+                DispatchQueue.main.async { completion(nil, defaultError) }
+                return
+            }
+
+            let jsonData = try? JSONSerialization.jsonObject(with: rawData)
+            guard let responseData = jsonData as? [String: Any] else {
+                DispatchQueue.main.async { completion(nil, defaultError) }
+                return
+            }
+            
+            DispatchQueue.main.async {
+                if "error" == responseData["status"] as? String {
+                    completion(nil, ApiError(response: responseData))
+                } else {
+                    completion(responseData, nil)
+                }
+            }
+        }.resume()
+    }
+    
+    static func urlWithQueryParameters(baseUrl: String, endpoint: String, parameters: [String: Any]) -> URL? {
+        let parametersDict: [String: String] = {
+            var parametersDict: [String: String] = [:]
+            for (key, value) in parameters {
+                if let val = value as? String {
```

Comment:
I think what you want here is to replace this whole `if` block with:

`let val = "\(value)"`

The current implementation still silently ignores non string values, which I think we should avoid. My suggestion will at least convert everything to a string, but I can also see the case for erroring out here and requiring the caller to set it.

Reply:
Also, even with this I'd still be careful to check for arrays / dictionaries / optionals  and error out if you get something that you're not expecting.